### DISCUSSION TITLE
.Net c#

### DISCUSSION THREAD

**Discussion by alentor** [gorules/zen#206] [Q&A]
Hello,

I'm wondering if there are any plans to support dotnet/c#. If there are, I'd be interested in contributing in any way I can.

Thank you.

**Comment by stefan-gorules** [comment]
Hi @alentor,

Dotnet is on our radar and there is some work done for it already. We'd be more than happy to get some help/expertise particularly with how .dlls would be shipped for each architecture.

At the moment we don't have capacity for it, but I can create a WIP repository for C# and we can start from there. 

Let me know how that sounds.

**Reply by alentor** [reply]
I'd like to help!

I'm not worried about the way .dll will be shipped, as you can include a .dll as part of your nugget package.

First I'd say we start to implement c# for Linux and Windows for x64 arch.

I have zero knowledge in Rust, have no objection in learning it.
From a brief reading it's possible to call Rust functions form C# but we might need to modify the functions in order for them to be callable from c#.

I'd start the project by putting build and runtime constrains for our currently supported platforms and moving from there to actual implementation..

**Reply by stefan-gorules** [reply]
Hi @alentor, please have a look at https://github.com/gorules/zen-dotnet/. Most of the implementation itself is already done.

Parts left are:
- Compiling dotnet binaries (see branch experiments/ffi on gorules/zen)
- Publishing to nuget (see https://github.com/gorules/zen-dotnet)

This particular version worked fine locally but was missing other architectures and some testing. Appreciate the effort, let me know if there's anything I might be able to assist with.
